jvector-base/src/main/java/io/github/jbellis/jvector/graph/OnHeapGraphIndex.java

```diff
@@ -366,47 +366,56 @@ public class ConcurrentGraphIndexView extends FrozenView {
 
         @Override
         public NodesIterator getNeighborsIterator(int level, int node) {
-            var it = getNeighbors(level, node).iterator();
-            return new NodesIterator() {
-                int nextNode = advance();
-
-                private int advance() {
-                    while (it.hasNext()) {
-                        int n = it.nextInt();
-                        if (completions.completedAt(n) < timestamp) {
-                            return n;
+            NodesIterator it;
+            try {
+                it = getNeighbors(level, node).iterator();
```

Comment:
Instead of using exceptions as control flow for an expected event, let's branch based on whether the result of getNeighbors is null.